Extensions/DataSources/src/DataSourcesHub/action-creators/DataSourcesActionCreators.tsx

```diff
@@ -0,0 +1,195 @@
+            serviceEndpointRestClient.executeServiceEndpointRequest(serviceEndpointReq, project, endpointId).then((result: ServiceEndpointRequestResult) => {
+                var payload: PayloadForExecuteServiceEndpointRequest = {
+                    result: result,
+                    executeError: null,
+                    parseError: null
+                }
+                DataSourcesActions.ExecuteServiceEndpointRequest(payload);
+
+            }).catch((error) => {
+                var payload: PayloadForExecuteServiceEndpointRequest = {
+                    result: null,
+                    parseError: null,
+                    executeError: {
+                        name: error['name'],
+                        status: error['status'],
+                        responseSheet: error['responseSheet'],
+                        message: error['message']
+                    }
+                }
+                DataSourcesActions.ExecuteServiceEndpointRequest(payload);
+            });
+        }
+    }
+
+    public getDataSources() {
+        var endpointId = this.getEndpointId();
+        var project = this.getProject();
+        var serviceEndpointRestClient = getClient(ServiceEndpointRestClient);
+        var data: DataSourceInfo = {}
+        if(endpointId!=null && project!=null){
+            serviceEndpointRestClient.getServiceEndpointDetails(project, endpointId).then((endpointdetails: ServiceEndpointDetails) => {
+                serviceEndpointRestClient.getServiceEndpointTypes(endpointdetails.type).then((typeDetails: ServiceEndpointType[]) => {
+                    if (typeDetails.length == 1) {
+                        var dataSources: DataSource[] = typeDetails[0].dataSources;
+                        if (dataSources != null && dataSources != undefined){
+                            for (let key of Object.keys(dataSources))
+                                data[dataSources[key].name] = dataSources[key]
+                        }
+                        DataSourcesActions.GetDataSources(data,endpointdetails);    
+                    }
+                    else{
+                        DataSourcesActions.GetDataSources(undefined,endpointdetails);
+                    }
+                });
+            });
+        }  
+    }
+
+    private getProject(): string| null {
+        var currentUrl = document.referrer;
+        var regex_details = /\/\/(.*)\/_settings/;
+        var detailsMatch = (currentUrl.match(regex_details));
+        var project=null;
+        if(detailsMatch!=null && detailsMatch.length==2){
+            var details=detailsMatch[1];
+            var projectMatch = (details.split("/"));
+            if(projectMatch!=null && projectMatch.length==3){
+                      project=projectMatch[2]
+            }
+        }
+        return project;
+    }
+
+    private extractParameters(datasourceInfo: string,currentInputParam:ParamInfo | null):ParamInfo{
+        var Parameters: ParamInfo = {}
```

Comment:
this method is little hard to understand.
lets sit together and see how can we make it easier